### PR TITLE
Refactor/new menu item

### PR DESCRIPTION
**Pull request by lucasbordeau**
Replaced DropdownMenuItem, DropdownMenuCheckableItem and DropdownMenuSelectableItem with new MenuItem, MenuItemSelect and MenuItemMultiSelect

### CODE REVIEW

File: front/src/modules/activities/components/ActivityEditor.tsx [hunk 1]
```diff
@@ -185,7 +185,7 @@ export function ActivityEditor({
               <>
                 <DateEditableField
                   value={activity.dueAt}
-                  icon={<IconCalendar />}
+                  Icon={IconCalendar}
```
Comment (charlesBochet) [inline front/src/modules/activities/components/ActivityEditor.tsx:188]: we also need to put this in the doc: when a props is a component, we must write it in PascalCase

File: front/src/modules/people/utils/fieldsForPerson.tsx [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import { isValidPhoneNumber } from 'libphonenumber-js';
 
+import { Fields } from '@/spreadsheet-import/types';
```
Comment (charlesBochet) [inline front/src/modules/people/utils/fieldsForPerson.tsx:3]: is it the wright import? I'm surprised that fieldsForPerson would depend on spreadheet types

File: front/src/modules/spreadsheet-import/types/index.ts [hunk 3]
```diff
@@ -96,7 +97,7 @@ export type Select = {
 
 export type SelectOption = {
   // Icon
-  icon?: React.ReactNode;
+  icon?: IconComponent | null | undefined;
```
Comment (charlesBochet) [inline front/src/modules/spreadsheet-import/types/index.ts:100]: `| undefined` is redundant

File: front/src/modules/ui/board/components/BoardColumnEditTitleMenu.tsx [hunk 4]
```diff
@@ -57,7 +47,10 @@ export const COLOR_OPTIONS = [
   { name: 'Orange', id: 'orange' },
   { name: 'Yellow', id: 'yellow' },
   { name: 'Gray', id: 'gray' },
-];
+] as {
```
Comment (charlesBochet) [inline front/src/modules/ui/board/components/BoardColumnEditTitleMenu.tsx:50]: do we need to use `as` keyword here we should be able to use COLOR_OPTIONS: { name: string, id: ThemeColor }

File: front/src/modules/ui/theme/constants/colors.ts [hunk 5]
```diff
@@ -22,6 +22,8 @@ export const grayScale = {
   gray0: '#ffffff',
 };
 
+export type ThemeColor = keyof typeof color;
```
Comment (charlesBochet) [inline front/src/modules/ui/theme/constants/colors.ts:25]: actually, the only ThemeColor that we should be able to use are the "primary" ones:  yellow | green... and not yellowXXX

File: front/src/modules/ui/menu-item/components/MenuItem.tsx [hunk 6]
```diff
@@ -10,21 +11,21 @@ import { MenuItemAccent } from '../types/MenuItemAccent';
...
 export type MenuItemIconButton = {
   Icon: IconComponent;
-  onClick: () => void;
+  onClick?: (event: MouseEvent<any>) => void;
 };
 
 export type MenuItemProps = {
-  LeftIcon?: IconComponent;
-  accent: MenuItemAccent;
+  LeftIcon?: IconComponent | null | undefined;
```
Comment (charlesBochet) [inline front/src/modules/ui/menu-item/components/MenuItem.tsx:18]: `| undefined` seems redundant

File: front/src/modules/ui/menu-item/components/MenuItemMultiSelectAvatar.tsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
+import { ReactNode } from 'react';
```
Comment (charlesBochet) [inline front/src/modules/ui/menu-item/components/MenuItemMultiSelectAvatar.tsx:1]: we should add stories on this one

File: front/src/modules/ui/menu-item/components/MenuItemSelectAvatar.tsx [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import { ReactNode } from 'react';
```
Comment (charlesBochet) [inline front/src/modules/ui/menu-item/components/MenuItemSelectAvatar.tsx:1]: here too we should add stories

File: front/src/modules/ui/menu-item/internals/components/MenuItemLeftContent.tsx [hunk 9]
```diff
@@ -1,14 +1,15 @@
...
+import { OverflowingTextWithTooltip } from '@/ui/tooltip/OverflowingTextWithTooltip';
 
 import {
   StyledMenuItemLabel,
   StyledMenuItemLeftContent,
 } from './StyledMenuItemBase';
 
 type OwnProps = {
-  LeftIcon?: IconComponent;
+  LeftIcon?: IconComponent | null | undefined;
```
Comment (charlesBochet) [inline front/src/modules/ui/menu-item/internals/components/MenuItemLeftContent.tsx:12]: undefined redundant

File: front/src/modules/companies/hooks/useUpdateCompanyBoardColumns.ts [hunk 10]
```diff
@@ -100,7 +101,7 @@ export function useUpdateCompanyBoard() {
           orderedPipelineStages?.map((pipelineStage) => ({
             id: pipelineStage.id,
             title: pipelineStage.name,
-            colorCode: pipelineStage.color,
+            colorCode: pipelineStage.color as ThemeColor,
```
Comment (charlesBochet) [inline front/src/modules/companies/hooks/useUpdateCompanyBoardColumns.ts:104]: I'm a bit annoyed by all these `as` casting. Is there a way to replace that by the right string literal in graphql? or to perform run-time guards?
We have avoided putting as much `as` as possible in the codebase so far, I feel this is opening the deamons gate!

**Review by charlesBochet — COMMENTED** [review]
Big piece!

File: front/src/modules/people/utils/fieldsForPerson.tsx (hunk 2 above)
Comment (lucasbordeau) [reply front/src/modules/people/utils/fieldsForPerson.tsx:3]: Yes it is used in `useSpreadsheetPersonImport`

File: front/src/modules/companies/hooks/useUpdateCompanyBoardColumns.ts (hunk 10 above)
Comment (lucasbordeau) [reply front/src/modules/companies/hooks/useUpdateCompanyBoardColumns.ts:104]: GraphQL codegen generates TS enum on top of GraphQL enums, the best approach is to use a typeguard yes !

File: front/src/modules/ui/menu-item/components/MenuItemMultiSelectAvatar.tsx (hunk 7 above)
Comment (lucasbordeau) [reply front/src/modules/ui/menu-item/components/MenuItemMultiSelectAvatar.tsx:1]: Created a ticket for this : https://github.com/twentyhq/twenty/issues/1464

**Review by charlesBochet — APPROVED** [review]
LGTM, thanks @lucasbordeau!

**Pull request merged**